zetaclient/chains/solana/observer/outbound.go

```diff
@@ -291,13 +291,33 @@ func ParseGatewayInstruction(
 		return nil, errors.Wrap(err, "error unmarshaling transaction")
 	}
 
-	// there should be only one single instruction ('withdraw' or 'withdraw_spl_token')
-	if len(tx.Message.Instructions) != 1 {
-		return nil, fmt.Errorf("want 1 instruction, got %d", len(tx.Message.Instructions))
+	// validate instruction count
+	// if there are 2 instructions, first one can only be optional compute budget instruction
+	instructionCount := len(tx.Message.Instructions)
```

Comment:
Nit: if you like, make a switch case on `len(tx.Message.Instructions)` would look very nice, there are 3 cases: 1, 2 and default. It's up to you though.